samples/BGV_world_country_db_lookup/BGV_world_country_db_lookup.cpp

```diff
@@ -0,0 +1,378 @@
+/*
+* MIT License
+*
+* Copyright (c) 2020 International Business Machines
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+// This is a sample program for education purposes only.
+// It implements a very simple homomorphic encryption based
+// db search algorithm for demonstration purposes.
+
+// This country lookup example is derived from the BGV database demo
+// code originally writte by Jack Crawford for a lunch and learn
+// session at IBM Research (Hursley) in 2019.
+// The original example code ships with HElib and can be found at
+// https://github.com/homenc/HElib/tree/master/examples/BGV_database_lookup
+
+#include <iostream>
+
+#include "helayers/hebase/hebase.h"
+#include "helayers/hebase/helib/HelibBgvContext.h"
+#include <fstream>
+#include <helib/ArgMap.h>
+#include <NTL/BasicThreadPool.h>
+#include <cassert>
+
+using namespace helayers;
+using namespace std;
+
+// Forward declarations. These functions are explained later.
+vector<pair<string, string>> read_csv(string filename, int maxLen);
+void run(HeContext& he, const string& db_filename, bool debug);
+bool isPowerOf2(int v);
+vector<int> stringToAscii(const string& val);
+
+int main(int argc, char* argv[])
+{
+  // Note: The parameters have been chosen to provide a somewhat
+  // faster running time with a non-realistic security level.
+  // Do Not use these parameters in real applications.
+
+  // Plaintext prime modulus
+  unsigned long p = 127;
```

Comment:
It would be great if we could re-work some of this, to make it clearer for someone who wants to better understand it to use for their own project.  It seems like we are using a specific configuration set, so maybe its worth explaining why we are choosing these values and how they affect everything.  Also this is something that we should blanket apply to all of the samples